utils/auth.py

```
@@ -0,0 +1,124 @@
+                "grant_type": "urn:ietf:params:oauth:grant-type:device_code",
+                "device_code": device_code,
+                "client_id": client_id,
+                "client_secret": client_secret,
+            },
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+            timeout=10,
+        )
+        if token_response.status_code == 200:
+            return token_response.json()
+        await asyncio.sleep(polling_interval_seconds)
+    raise Exception("Timeout waiting for authorization.")
```

Ruff (0.14.10)

Create your own exception

(TRY002)